lib/k8y/rest/auth/providers/gcp/factory.rb

```diff
@@ -11,21 +11,17 @@ module GCP
           Error = Class.new(Error)
 
           class Factory
-            MissingConfigError = Class.new(Error)
-
             def from_auth_provider(provider)
-              config = provider[:config]
-              raise MissingConfigError unless config
-
+              config = provider.config
               # see https://github.com/kubernetes/client-go/blob/master/plugin/pkg/client/auth/gcp/gcp.go#L58
-              if config[:"cmd-path"]
+              if config&.public_send(:"cmd-path")
                 CommandProvider.new(
```

Comment:
config might actually have a token already. However, it's straightforward to assume we need to generate one. Since we'll be using tokenstore and not making additional needless calls, the simplification in code might be worth the extra request